### DISCUSSION TITLE
materials.GetListMaterialProperties

### DISCUSSION THREAD

**Discussion by Dirk-Haverkamp** [ansys/pymechanical#847] [Q&A] [answered]
Hello,

with ACT I was able to write something like: 

```
import material 
...
mat = body.Material
print materials.GetListMaterialProperties(mat)
```

Is the method "GetListMaterialProperties" also accessible in pyMechanical?

Best regards,
DIrk

**Comment by dipinknair** [comment, marked as answer]
Yes. This is accessible through ``add_mechanical_python_libraries`` function.  Here is how to use it.

Start pymechanical app
```
from ansys.mechanical.core import App
app=App(version=241)  #version may vary
app.update_globals(globals())
print(app)
```
import function and materials
```
from ansys.mechanical.core.embedding.app_libraries import add_mechanical_python_libraries
add_mechanical_python_libraries(app)

import materials
```
import geometry and materials needed then call required function from materials

```
mat = body.Material
print(materials.GetListMaterialProperties(mat))
```

ref: https://mechanical.docs.pyansys.com/version/stable/user_guide_embedding/libraries.html

**Reply by Dirk-Haverkamp** [reply]
Thanks for the help. 
The following code creates an error:
```
from ansys.mechanical.core import App
from ansys.mechanical.core.embedding import add_mechanical_python_libraries
app = App(version=241)
app.update_globals(globals())
print(app)

add_mechanical_python_libraries(app)
import materials
app.open(r"D:\HAD5SI\30_smarttc\timestampfolder_20240801134815\work\smarttc_files\dp0\global\MECH\SYS.mechdb")
body = ExtAPI.DataModel.GeoData.Assemblies[0].Parts[0].Bodies[0]
mat = body.Material
print(materials.GetListMaterialProperties(mat))
```

```
CRITICAL -  -  logging - handle_exception - Uncaught exception
Traceback (most recent call last):
  File "d:\HAD5SI\30_smarttc\repo\smarttc_pyansys\src\smarttc\mechanical\test.py", line 13, in <module>
    print(materials.GetListMaterialProperties(mat))
    list.Add(prop.TypeName)
AttributeError: 'list' object has no attribute 'Add'
```

I didn't remember that "materials" is a library that is in the installation: C:\Program Files\ANSYS Inc\v241\Addins\ACT\libraries\Mechanical\materials.py
So I can look at the code and write my own function:

```
  def GetListMaterialProperties(mat):
      """
      Return the list of material property names for the given material. 
      """
      list = []
      for prop in mat.MaterialProperties:
          list.append(prop.TypeName)
      return list

```
Now I have a solution that works for me. Thank you

**Reply by dipinknair** [reply]
Glad that it worked !

**Reply by koubaa** [reply]
@Dirk-Haverkamp it will work in 2024R2.  The code in `materials.py` was only compatible with IronPython in 2024 R1, but in 2024R2 it was made compatible with both IronPython and CPython.
